discovery-provider/integration_tests/tasks/test_anchor_program_indexer.py

```diff
@@ -0,0 +1,151 @@
+import asyncio
+from unittest.mock import create_autospec
+
+from construct import ListContainer
+from src.models.models import AudiusDataTx
+from src.solana.anchor_program_indexer import AnchorProgramIndexer
+from src.solana.solana_client_manager import SolanaClientManager
+from src.utils.config import shared_config
+from src.utils.db_session import get_db
+from src.utils.redis_connection import get_redis
+from web3 import Web3
+
+PROGRAM_ID = shared_config["solana"]["anchor_data_program_id"]
+LABEL = "test_indexer"
+
+
+def test_exists_in_db(app):  # pylint: disable=W0621
+    with app.app_context():
+        db = get_db()
+        redis = get_redis()
+
+    solana_client_manager_mock = create_autospec(SolanaClientManager)
+    anchor_program_indexer = AnchorProgramIndexer(
+        PROGRAM_ID, LABEL, redis, db, solana_client_manager_mock
+    )
+
+    TEST_TX_HASH = "3EvzmLSZekcQn3zEGFUkaoXej9nUrwkomyTpu9PRBaJJDAtzFQ3woYuGmnLHrqY6kZJtxamqCgeu17euyGp3EN4W"
+    TEST_TX_SLOT = 100
+
+    with db.scoped_session() as session:
+        assert anchor_program_indexer.is_tx_in_db(session, TEST_TX_HASH) == False
+        session.add(AudiusDataTx(signature=TEST_TX_HASH, slot=TEST_TX_SLOT))
+        assert anchor_program_indexer.is_tx_in_db(session, TEST_TX_HASH) == True
+
+    latest_slot = anchor_program_indexer.get_latest_slot()
```

Comment:
this tests that anchor program indexer updates latest slot and is able to retrieve db status?

Reply:
exactly